lib/facter/simplib_networkmanager.rb

```diff
@@ -0,0 +1,49 @@
+Facter.add(:simplib_networkmanager) do
+  confine :kernel => 'Linux'
+
+  @nmcli_cmd = Facter::Util::Resolution.which('nmcli')
+  confine { @nmcli_cmd }
+
+  setcode do
+    info = { 'enabled' => false }
+
+    nmcli_cmd = @nmcli_cmd + ' -t'
+
+    general_status = Facter::Core::Execution.execute(%(#{nmcli_cmd} -m multiline general status), :on_fail => :failed)
+    unless general_status == :failed
+      general_status = general_status.lines.map{|line| line.strip.split(':') }
+
+      info['enabled'] = true
+      info = {
+        'general' => {
+          'status' => Hash[general_status]
+        }
+      }
+    end
+
+    general_hostname = Facter::Core::Execution.execute(%{#{nmcli_cmd} general hostname}, :on_fail => :failed)
+    unless general_hostname == :failed
+      info['enabled'] = true
+      info['general'] ||= {}
+      info['general']['hostname'] = general_hostname.strip
+    end
+
+    connections = Facter::Core::Execution.execute(%(#{nmcli_cmd} connection show), :on_fail => :failed)
+    unless connections == :failed
+      info['enabled'] = true
+      info['connection'] = {}
+
+      connections.lines.each do |conn|
+        name, uuid, type, device = conn.strip.split(':')
```

Comment:
Will this parsing will do what you expect on the output from my dev machine? Notice the spaces such as 'Bridge br1':

$myhost: nmcli -t connection show
Bridge br1:2ee981ca-5ff4-4f9b-03fe-32879aa3dc85:bridge:br1
docker0:17f50a67-7276-4c44-b47c-4c3c58fea91b:bridge:docker0
System enp0s31f6:b5f59873-5bd2-c5a6-acb6-f03d5581aca5:802-3-ethernet:enp0s31f6
virbr0:dbdb93ed-c297-424a-9398-0c16700758c6:bridge:virbr0

Reply:
> Do you think the names with spaces in them will be useful?

@lnemsick-simp The "Bridge br1" string is the Connection Profile's [`id` string](https://developer.gnome.org/NetworkManager/stable/nm-settings.html), which is a "readable unique identifier for the connection" that is encouraged to contain spaces.  NetworkManager auto-generates the id (with the spaces) unless you change it yourself, so we can expect the spaces to be present most of the time.  

The `id` names are "useful" with `nmcli connection COMMAND <ID>` commands to serve as the `<ID>`.   From what I've seen, the `<ID>` in all `nmcli con` commands will also accept `uuid` or (DBUS) `path`.